Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -90,10 +120,25 @@ public struct FindOneAndDeleteOptions: Encodable {
 
     /// An optional `WriteConcern` to use for the command.
     public let writeConcern: WriteConcern?
+
+    fileprivate func asOpts() throws -> FindAndModifyOptions {
+        return try FindAndModifyOptions(collation: collation, maxTimeMS: maxTimeMS, projection: projection,
+                                        remove: true, sort: sort, writeConcern: writeConcern)
+    }
+
+    /// Convenience initializer allowing any/all parameters to be omitted/optional
+    public init(collation: Document? = nil, maxTimeMS: Int64? = nil, projection: Document? = nil, sort: Document? = nil,
```

Comment:
really wish we had a way to auto-generate these initializers.... seems like it would be easy enough to build into Swift as some protocol using reflection

Reply:
According to https://medium.com/@sergueivinnitskii/easy-struct-initialization-in-swift-8ee46b8d84d5, structs do get a default initializer; however, an explicit initializer is required if you want the convenience of optional properties defaulting to `nil` if unspecified. I suppose that's what you're pining for here? :)

Reply:
yup, that's what I meant - we have these for every single options struct and they feel very boilerplate